### PR TITLE
Misc. documentation improvements

### PR DESCRIPTION
**Pull request by brianhelba**
These are some documentation improvements that I've built up locally over time. See the individual commit messages for details.

Marked WIP because I haven't had a chance to build the changes locally yet and look at the HTML output.

### CODE REVIEW

**Comment by brianhelba** [comment]
Pushed one more commit on top.

File: girder/api/rest.py [hunk 1]
```diff
@@ -20,13 +20,11 @@
 import cherrypy
 import collections
 import datetime
-import functools
 import json
 import pymongo
 import six
 import sys
 import traceback
-import types
```
Comment (zachmullen) [inline girder/api/rest.py]: Weird, our style checker usually catches unused imports, do you know why it didn't here?

**Comment by brianhelba** [comment]
Rebased and fixed an issue. Ready for review / merge.

File: girder/api/rest.py (hunk 1 above)
Comment (brianhelba) [inline girder/api/rest.py]: No idea. I was wondering this myself.

File: girder/api/rest.py [hunk 2]
```diff
@@ -39,6 +37,22 @@
...
+    Calls `urllib.parse.urlparse`_ on a URL.
+
+    Note: This is compatible with both Python 2 and 3.
+
+    :param url: A URL, or None to use the current request's URL.
+    :type url: str or None
+    :return: The URL's seperate components.
+    :rtype: `urllib.parse.ParseResult`_
+
+    .. _urllib.parse.urlparse:
```
Comment (brianhelba) [inline girder/api/rest.py]: This isn't generating properly. Working to fix it now.

**Comment by brianhelba** [comment]
Everything should be working now.

Pushed one small new commit on top too.

**Comment by brianhelba** [comment]
Added another commit fixing another small issue.

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**